extensions/vscode/src/multiStepInputs/newDeployment.ts

```diff
@@ -770,125 +350,222 @@ export async function newDeployment(
     // as long as we don't know it will take another one.
     inspectionResults = [];
 
-    // show only if we have more than one potential entrypoint discovered
-    if (discoveredEntryPoints.length > 1) {
-      const step = getStepInfo("inputEntryPointFileSelection", state);
-      if (!step) {
-        window.showErrorMessage(
-          "Internal Error: newDeployment::inputEntryPointFileSelection step info not found.",
-        );
-        return;
+    // show only if we were not passed in a file
+    if (entryPointFile === undefined) {
+      if (newDeploymentData.entrypoint.filePath) {
+        entryPointListItems.forEach((item) => {
+          item.picked = item.label === newDeploymentData.entrypoint.filePath;
+        });
       }
 
       const pick = await input.showQuickPick({
         title: state.title,
-        step: step.step,
-        totalSteps: step.totalSteps,
+        step: 0,
+        totalSteps: 0,
         placeholder:
           "Select entrypoint file. This is your main file for your project. (Use this field to filter selections.)",
         items: entryPointListItems,
         buttons: [],
         shouldResume: () => Promise.resolve(false),
         ignoreFocusOut: true,
       });
-      state.data.entryPointPath = pick.label;
+
+      let selectedEntrypointFile: string | undefined = undefined;
+      if (pick.label === browseForEntrypointLabel) {
+        let baseUri = Uri.parse(".");
+        const workspaceFolders = workspace.workspaceFolders;
+        if (workspaceFolders !== undefined) {
+          baseUri = workspaceFolders[0].uri;
+        }
+        selectedEntrypointFile = undefined;
+        do {
+          const fileUris = await window.showOpenDialog({
+            defaultUri: baseUri,
+            openLabel: "Select",
+            canSelectFolders: false,
+            canSelectMany: false,
+            title: "Select Entrypoint File (main file for your project)",
+          });
+          if (!fileUris || !fileUris[0]) {
+            // cancelled.
+            return;
+          }
```

Comment:
I found it a bit odd that canceling out of the show open dialog cancelled the entire flow. I'm not saying this _needs_ to change, perhaps others have differing opinions. For me I'd prefer if the multi-stepper stayed open and I had to dismiss it with `ESC` as normal.

Reply:
Fair enough. This has been updated.